lib/transformers/define_action_workers.ex

```diff
@@ -100,7 +100,7 @@ defmodule AshOban.Transformers.DefineActionWorkers do
 
               unquote(resource)
               |> Ash.ActionInput.new()
-              |> Ash.ActionInput.set_context(%{private: %{ash_oban?: true}})
+              |> Ash.ActionInput.set_context(%{shared: %{private: %{ash_oban?: true}}})
```

Comment:
I don't think we should do this. We can't guarantee that all actions called within an Oban job want to "think" they are happening in an Oban job. That's something for the person writing the action to decide IMO. We could make this opt-in with a flag?